OnBoardingCalculator/app/src/main/java/com/globant/onboardingcalculator/mvp/presenter/CalculatorPresenter.java

```diff
@@ -20,18 +33,74 @@ public void onClearPressed() {
     }
 
     public void onOperatorPressed(char operator) {
-        if (model.getOperator() == Constants.EMPTY_CHAR)
+        if (!model.getFirstOperand().isEmpty()) {
             model.setOperator(operator);
-        view.refreshVisor(String.valueOf(operator));
+            view.refreshVisor(String.valueOf(operator));
+            view.enablePointBtn();
+        } else
+            view.showOperatorError();
     }
 
     public void onNumberPressed(String number) {
-        if (model.getOperator() == Constants.EMPTY_CHAR) {
+        if (!model.getResult().equals(EMPTY_STRING) && (model.getOperator() == EMPTY_CHAR))
```

Comment:
Pls, use the method isEmpty to check this condition .

Reply:
Done! Thanks